go/interpreter/lfvm/instructions.go

```diff
@@ -1119,44 +1127,64 @@ func opCall(c *context) {
 		cost += tosca.Gas(params.CallStipend)
 	}
 
-	// If we are in static mode, recursive calls are to be treated like
-	// static calls. This is a consequence of the unification of the
-	// interpreter interfaces of EVMC and Geth.
-	// This problem was encountered in block 58413779, transaction 7.
-	kind := tosca.Call
-	if c.params.Static {
-		kind = tosca.StaticCall
-
-		// In a static call, no value must be transferred.
+	if kind == tosca.Call || kind == tosca.CallCode {
```

Comment:
Because static calls must have value 0. Could we drop this check and keep `if !value.IsZero() {` only?

Reply:
this check is to filter out delegate calls, but I will join it with if right under.